### PR TITLE
Remove `:confirmable` from our User model.

### PR DESCRIPTION
**Pull request by ashercodeorg**

### CODE REVIEW

**Comment by ashercodeorg** [comment]
Right, this PR isn't quite ready for review, I have more stuff to fix (the current CI failures are very much related).

**Comment by ashercodeorg** [comment]
Now ready for review. Please take another look.

File: dashboard/app/views/devise/mailer/confirmation_instructions.html.erb [hunk 1]
```diff
@@ -2,8 +2,6 @@
 
 <p>You can confirm your account email through the link below:</p>
 
-<p><%= link_to 'Confirm my account', confirmation_url(@resource, :confirmation_token => @token) %></p>
-
```
Comment (wjordan) [inline dashboard/app/views/devise/mailer/confirmation_instructions.html.erb:6]: should we remove this view altogether? I'm assuming it's no longer used at all, but if it is unintentionally referenced somewhere, i think we'd prefer to report an error than send a user an email missing a link.

Comment (ashercodeorg) [reply dashboard/app/views/devise/mailer/confirmation_instructions.html.erb:6]: This is attempting to do the minimal amount of work to remove `:confirmable` from the User model. This view, as well as other stuff, is being removed in #14512.

**Pull request merged**